server/api/src/rsGeneralAdmin.cpp

```diff
@@ -25,8 +27,142 @@
+        }
+        if ( ( timestamp = getSqlResultByInx( gen_out, COL_META_RESC_ATTR_UNITS ) ) == nullptr ) {
+            rodsLog( LOG_ERROR, "%s: getSqlResultByInx for COL_META_RESC_ATTR_UNITS failed", __FUNCTION__ );
+            return UNMATCHED_KEY_OR_INDEX;
+        }
+        std::stringstream msg;
+        msg << "A rebalance_operation on resource [";
+        msg << _resource_name.c_str();
+        msg << "] is still active (or stale) [";
+        msg << &hostname_and_pid->value[0];
+        msg << "] [";
+        msg << &timestamp->value[0];
+        msg << "]";
+        rodsLog( LOG_ERROR, "%s: %s", __FUNCTION__, msg.str().c_str() );
+        addRErrorMsg( &_rsComm->rError, REBALANCE_ALREADY_ACTIVE_ON_RESOURCE, msg.str().c_str() );
+        return REBALANCE_ALREADY_ACTIVE_ON_RESOURCE;
+    }
+    freeGenQueryOut( &gen_out );
+    clearGenQueryInp( &gen_inp );
+    return 0;
+}
+
+int _set_rebalance_timestamp_avu_on_resource(
+    rsComm_t* _rsComm,
+    const std::string& _resource_name) {
+
+    // get hostname
+    char hostname[MAX_NAME_LEN];
+    gethostname(hostname, MAX_NAME_LEN);
+    // get PID
+    int pid = getpid();
+    // get timestamp in defined format
+    const boost::posix_time::ptime timeUTC = boost::posix_time::second_clock::universal_time();
+    std::stringstream stream;
+    boost::posix_time::time_facet* facet = new boost::posix_time::time_facet();
+    facet->format("%Y%m%dT%H%M%SZ");
+    stream.imbue(std::locale(std::locale::classic(), facet));
+    stream << timeUTC;
+    // add AVU to resource
+    modAVUMetadataInp_t modAVUMetadataInp;
+    memset( &modAVUMetadataInp, 0, sizeof( modAVUMetadataInp ) );
+    modAVUMetadataInp.arg0 = "set";
+    modAVUMetadataInp.arg1 = "-R";
+    modAVUMetadataInp.arg2 = strdup( _resource_name.c_str() );
+    modAVUMetadataInp.arg3 = "rebalance_operation";
+    std::string value = std::string(hostname) + std::string(":") + std::to_string(pid);
+    modAVUMetadataInp.arg4 = strdup( value.c_str() );
+    std::string unit = stream.str();
+    modAVUMetadataInp.arg5 = strdup( unit.c_str() );
+    // do it
+    return rsModAVUMetadata( _rsComm, &modAVUMetadataInp );
+}
+
+int _check_and_set_rebalance_timestamp_avu_on_resource(
+    rsComm_t* _rsComm,
+    const std::string& _resource_name) {
+
+    int status = _check_rebalance_timestamp_avu_on_resource(_rsComm, _resource_name);
+    if (status < 0 ){
+        return status;
+    }
+    status = _set_rebalance_timestamp_avu_on_resource(_rsComm, _resource_name);
+    if (status < 0 ){
+        return status;
+    }
+    return 0;
+}
+
+int _remove_rebalance_timestamp_avu_from_resource(
+    rsComm_t* _rsComm,
+    const std::string& _resource_name) {
+    // build genquery to find active or stale "rebalance operation" entries for this resource
+    genQueryOut_t* gen_out = nullptr;
+    int status = 0;
+    char tmp_str[MAX_NAME_LEN];
+    genQueryInp_t gen_inp;
+    memset( &gen_inp, 0, sizeof( genQueryInp_t ) );
```

Comment:
If `genQueryInp_t` is a POD type, then you can replace lines 137 and 138 with `genQueryInp_t gen_inp{}`.  The empty braces clear the struct just like `memset`.  This also applies to other POD types used throughout your PR (e.g. `modAVUMetadataInp_t`).